x-pack/plugins/observability_solution/apm/public/components/app/service_inventory/apm_signal_inventory/index.tsx

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { usePerformanceContext } from '@kbn/ebt-tools';
+import { EuiEmptyPrompt, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { apmEnableServiceInventoryTableSearchBar } from '@kbn/observability-plugin/common';
+import { useEditableSettings } from '@kbn/observability-shared-plugin/public';
+import { ApmDocumentType } from '../../../../../common/document_type';
+import {
+  ServiceInventoryFieldName,
+  ServiceListItem,
+} from '../../../../../common/service_inventory';
+import { useAnomalyDetectionJobsContext } from '../../../../context/anomaly_detection_jobs/use_anomaly_detection_jobs_context';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useStateDebounced } from '../../../../hooks/use_debounce';
+import { FETCH_STATUS, isFailure, isPending } from '../../../../hooks/use_fetcher';
+import { useLocalStorage } from '../../../../hooks/use_local_storage';
+import { usePreferredDataSourceAndBucketSize } from '../../../../hooks/use_preferred_data_source_and_bucket_size';
+import { useProgressiveFetcher } from '../../../../hooks/use_progressive_fetcher';
+import { useTimeRange } from '../../../../hooks/use_time_range';
+import { APIReturnType } from '../../../../services/rest/create_call_apm_api';
+import { SortFunction } from '../../../shared/managed_table';
+import { MLCallout, shouldDisplayMlCallout } from '../../../shared/ml_callout';
+import { SearchBar } from '../../../shared/search_bar/search_bar';
+import { isTimeComparison } from '../../../shared/time_comparison/get_comparison_options';
+import { ApmServicesTable } from './service_list/apm_services_table';
+import { orderServiceItems } from './service_list/order_service_items';
+
+type MainStatisticsApiResponse = APIReturnType<'GET /internal/apm/services'>;
+
+const INITIAL_PAGE_SIZE = 25;
+const INITIAL_DATA: MainStatisticsApiResponse & { requestId: string } = {
+  requestId: '',
+  items: [],
+  serviceOverflowCount: 0,
+  maxCountExceeded: false,
+};
+
+function useServicesMainStatisticsFetcher(searchQuery: string | undefined) {
```

Comment:
Should we move these hooks into their separate files? I got the impression that this file is too getting too long, wdyt?

Reply:
I tend to agree, the file is too long already but I think it's easier to read the code when the request it's co-located with the UI component. However, I didn't modify this component, I just renamed it thus I'd like to keep it as-is for now.